<launch>
  <!-- declare args to be passed in -->
  <arg name="obj" default='{"type" : "Cup"}'/> 
  <arg name="method" default='random'/> 
  <arg name="robot" default='real'/> 

  <!-- nav goal generation -->
  <param name="inflation_radius" type="double" value="0.6" />

  <!-- search params -->  
  <param name="obj_desc" type="string" value="$(arg obj)" />
  <param name="robot" type="string" value="$(arg robot)" />    
  <param name="search_method" type="string" value="$(arg method)" />    

  <!-- the polygon spefifies the search area in global map coordinates (default is full map) -->

  <!-- robot lab -->
  <rosparam param="search_area">[[1.2,0.5],[-4.6,0.05],[-3.5,-6.0],[1.5,-6.0]]</rosparam>

  <!-- robot lab + atrium -->
  <!-- <rosparam param="search_area">[[0,0],[2,-2],[-2,-2]]</rosparam> -->

  <param name="min_objs" type="int" value="1" />
  <param name="max_objs" type="int" value="1" />
  <param name="max_time" type="int" value="300" />
  <param name="max_poses" type="int" value="50" />  
  
  <node name="search_agent" pkg="object_search" type="search_agent.py" output="screen"/>
</launch>
